src/Command/AnonymizeUsersCommand.php

```diff
@@ -0,0 +1,165 @@
+    {
+        return parent::buildOptionParser($parser)
+            ->addOptions([
+                'id' => [
+                    'help' => 'User id',
+                ],
+                'preserve' => [
+                    'help' => 'Users to preserve by id',
+                    'default' => '1',
+                ],
+            ]);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Anonymize Users command.
+     *
+     * $ bin/cake anonymize_users --help
+     *
+     * Usage:
+     * cake anonymize_users [options]
+     *
+     * Options:
+     *
+     * --id           User id
+     * --preserve     Users to preserve by id
+     * --help, -h     Display this help.
+     * --verbose, -v  Enable verbose output.
+     *
+     * # basic
+     * $ bin/cake anonymize_users --id 2
+     * $ bin/cake anonymize_users --preserve 1,2,3
+     */
+    public function execute(Arguments $args, ConsoleIo $io)
+    {
+        $io->success('Start.');
+        LoggedUser::setUserAdmin();
+        /** @var \BEdita\Core\Model\Table\UsersTable $table */
+        $table = $this->fetchTable('Users');
+        $this->preserveUsers = array_map('intval', explode(',', $args->getOption('preserve')));
+        $query = $table->find()
+            ->where([
+                $table->aliasField('locked') => 0,
+                $table->aliasField('deleted') => 0,
+                $table->aliasField('id') . ' NOT IN' => $this->preserveUsers,
+            ]);
+        $id = $args->getOption('id');
+        if ($id) {
+            $query->where([$table->aliasField('id') => $id]);
+        }
+        $faker = Factory::create('it_IT');
+        $processed = $saved = $errors = 0;
+        /** @var \BEdita\Core\Model\Entity\User $user */
+        foreach ($this->objectsGenerator($query) as $user) {
+            $this->anonymize($faker, $user, $table, $io, $processed, $saved, $errors);
+        }
+        $io->out(sprintf('Users processed: %s', $processed));
+        $io->out(sprintf('Users saved: %s', $saved));
+        $io->out(sprintf('Users not saved: %s', $errors));
+        $io->success('Done.');
+
+        return null;
+    }
+
+    /**
+     * Update user.
+     *
+     * @param \Faker\Generator $faker Faker generator
+     * @param \BEdita\Core\Model\Entity\User $user User entity
+     * @param \BEdita\Core\Model\Table\UsersTable $table Users table
+     * @param \Cake\Console\ConsoleIo $io Console IO
+     * @param int $processed Processed users
+     * @param int $saved Saved users
+     * @param int $errors Errors
+     * @return void
+     */
+    public function anonymize(Generator $faker, User $user, UsersTable $table, ConsoleIo $io, int &$processed, int &$saved, int &$errors): void
+    {
+        $io->verbose(sprintf('Processing user %s [username: %s, email: %s]', $user->id, $user->username, $user->email));
+        $user->name = $faker->firstName();
+        $user->surname = $faker->lastName();
+        do {
+            $email = $faker->email();
```

Comment:
I'd rather use `safeEmail` instead ([docs](https://fakerphp.org/formatters/internet/#safeemail)), as it is guaranteed to generate an email address for example.com/.org, which are reserved domain names. This ensures that even if a real email is sent by the system, no real user will ever receive it.

```suggestion
            $email = $faker->safeEmail();
```